web/bfex/components/data_pipeline/tasks/keywords.py

```diff
@@ -4,68 +4,64 @@
 from bfex.common.exceptions import WorkflowException
 from bfex.components.key_generation.rake_approach import *
 from bfex.components.key_generation.generic_approach import *
+from bfex.components.key_generation.key_generator import KeyGenerator
+import time
 
-
-
-class UpdateKeywordsFromScrape(Task):
+class GetKeywordsFromScrape(Task):
     """
     Updates Keywords of a Faculty Members data in elastic.
     """
     def __init__(self):
         self.task_name = "Update Keywords From Scrape"
 
-    def is_requirement_satisfied(self, data):
+    def is_requirement_satisfied(self, faculty):
         satisfied = True
 
-        for faculty in data:
-
-            if (not isinstance(faculty, tuple) or
-                    not isinstance(faculty[0], str) or
-                    not isinstance(faculty[1], Scrapp)):
-                satisfied = False
+        if (not isinstance(faculty.faculty_id, int)):
+            satisfied = False
 
         return satisfied
 
 
-    def run(self,data):
-        """ Updates keywords of all profs
+    def run(self,faculty):
+        """ Updates keywords of a prof
 
-        :param data is a faculty object
-        :return: last faculty member handled
+        :param faculty is a faculty object
+        :return: list of keyword objects
         """
-        no_text_count = 0
-        for faculty in data:
-            faculty_name = faculty.name
-    
-            search_results = Faculty.search().query('match', name=faculty_name).execute()
-            if len(search_results) > 1:
-                # Shouldn't happen, but could.
-                raise WorkflowException("Professor id is ambiguous during search... More than 1 result")
-        
-            faculty = search_results[0]
-            if faculty.text != None:
-                
-                rake = RakeApproach()
-                rake_keyword = rake.generate_keywords(faculty.text)
-                faculty.rake_keywords = rake_keyword
+        key_objects= []
+        faculty_id = faculty.faculty_id
+        time.sleep(1)
```

Comment:
Is this to circumvent the data not saving? We might be better off adding a delay between tasks in Workflow

Reply:
yes, this is to give it enough to post to elasticsearch before indexing the data, a delay between tasks in workflow would probably be better.